src/cmap/establish/handshake/mod.rs

```diff
@@ -229,6 +244,32 @@ impl Handshaker {
                 .map(|server_first| client_first.into_first_round(server_first))
         });
 
+        // Check that master reply has a compressor list and unpack it
+        if let Some(ref server_compressors) = is_master_reply.command_response.compressors {
+            // Check that client has a compressor list and unpack it
+            if let Some(ref client_compressors) = self.compressors {
```

Comment:
I tend to bundle pattern matches like this via tuples, i.e.
```
if let (Some(ref server_compressors), Some(ref client_compressors)) = (is_master_reply.command_response.compressors, self.compressors) {
  ...
```

Reply:
Nice, will do.  That was bothering me.